src/test.js

```diff
@@ -272,12 +272,9 @@ Test.prototype = {
 				runtime: now() - this.started
 			};
 
-		if ( !result ) {
-			source = sourceFromStacktrace();
-
-			if ( source ) {
-				details.source = source;
-			}
+		source = sourceFromStacktrace( 4 );
+		if ( source ) {
+			details.source = source;
```

Comment:
As I mentioned before, the sourceFromStacktrace parameter must be different when the test fail, otherwise you're just gonna solve the problem when it pass, before it was good when the tests failed.